src/services/cache-service.ts

```diff
@@ -32,9 +32,12 @@ export class CacheService {
         this._cacheContainer[key] = [
             validUntil,
             loadingFunction().then(response => {
-                const cachedResponse = Promise.resolve(response);
-                this._cacheContainer[key] = [validUntil, cachedResponse];
-                return cachedResponse;
+                const cachedResponse = Promise.resolve(response)
+                if (validUntil >= Date.now()) { // lest older, dead key replace newer, still living one
+                    this._cacheContainer[key] = [validUntil, cachedResponse]
+                }
+                // outside 'if', so previous handles on this call still eventually receive the result
+                return cachedResponse
```

Comment:
I get your point: When a long running function takes longer than the cache ttl, it could replace an existing newer cache by an already outdated one.
But why is the test below so heavenly complex?

Reply:
Yes, certainly. It is how it is because I had to construct it.

E.g. the last line is the original point of failure, but much later I added in the middle the line checking for outdated keys, which will fail _in exactly the same circumstances_ .

Similary, call 2 never failed but was only there for, during construction, making sure that we'd not fail for other reasons.

... In comparison, I do not really think that from the (correctly) simplified version in #51 one can still reconstruct the actual user-faced behaviour and the expectations on it. But to be fair, I was rather unsure whether this would be of more-than-historical interest, so I approved of the simplification - it's basically you telling me: "No, we do not need that!"

The many Sanity Checks have also been motivated by whole three people missing the real bug one by one: you, me, @conradstrassburger all had a look and all thought it would be fine, until my, hè, "professional paranoia" led me to double-check even though I "already knew better".